File: leetcode-167-Two_Sum_II_-_Input_Array_Is_Sorted/leetcode-167-Two_Sum_II_-_Input_Array_Is_Sorted.cpp
#include <iostream>
#include <cstdlib>
#include <vector>
#include <string>


std::vector<int> twoSum(std::vector<int>& numbers, int target) {
    std::vector<int> res;

    size_t i = 0,
            j = (numbers.size()-1);
    while(i<j)
    {
        int cur_sum = (numbers[i]+numbers[j]);
        if(target<cur_sum)
        {
            --j;
        } else if(cur_sum<target) {
            ++i;
        } else {
            res.emplace_back(i+1);
            res.emplace_back(j+1);
            return res;
        }
    }
    return res;
}


int main(){
    std::vector<int> numbers = {2,7,11,15};
    int target = 9;

    auto res = twoSum(numbers, target);
    
    std::cout << "[";
    for (size_t i = 0; i<res.size(); ++i)
    {
        std::cout << res[i];
        if(i!=(res.size()-1))
        {
            std::cout << ",";
        }
    }
    std::cout << "]" << std::endl;
    
    return EXIT_SUCCESS;
}

File: leetcode-167-Two_Sum_II_-_Input_Array_Is_Sorted/leetcode-167-Two_Sum_II_-_Input_Array_Is_Sorted.py
class Solution:
    def twoSum(self, numbers: list[int], target: int) -> list[int]:
        left: int = 0
        right: int = len(numbers) - 1
        while left < right:
            curr: int = numbers[left] + numbers[right]
            if curr == target:
                break
            elif curr < target:
                left += 1
            else:
                right -= 1
        return [left+1, right+1]


if __name__ == "__main__":
    numbers: list[int] = [2,7,11,15]
    target: int = 9
    result = Solution().twoSum(numbers, target)
    print(result)
